compiler/src/vm/mod.rs

```diff
@@ -473,10 +438,13 @@ impl Vm {
             let Parallel {
                 mut paused_fiber,
                 nursery,
+                return_value,
                 ..
             } = tree.into_parallel().unwrap();
             self.channels.remove(&nursery).unwrap();
-            paused_fiber.fiber.complete_parallel_scope(result);
+            paused_fiber
+                .fiber
+                .complete_parallel_scope(result.map(|_| return_value.unwrap()));
```

Comment:
It looks like `result` and `return_value` are partial duplicates. Around line 368, if the fiber completes successfully, the return value is stored in the fiber and then this function is called with `Ok(())`. Can we instead remove `parallel.return_value` and pass the value to this function directly?

Reply:
I first thought about doing that, but it may be that the first fiber exits way before the last fiber of the parallel section.

```candy
core.parallel { nursery ->
  core.async {
    # long running computation
  }
  Foo
}
```
 
In this example, the initial child returns `Foo` while the spawned fiber is still running.

Reply:
Oh, that makes sense